x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/operator/arithmetic/Mod.java

```diff
@@ -30,4 +30,10 @@ public Mod(Source source, Expression left, Expression right) {
     protected Mod replaceChildren(Expression newLeft, Expression newRight) {
         return new Mod(source(), newLeft, newRight);
     }
+
+    @Override
+    public ArithmeticOperation inverse(Source source, Expression left, Expression right) {
+        // TODO: Modular Multiplicative Inverse, if ever needed?
+        throw new UnsupportedOperationException("inverting modulo operation is not supported");
+    }
```

Comment:
The exception type is wrong. And we rarely thrown an exception on methods that are forced to be implemented - that indicates either the code is incomplete (as here) or that the contract is too broad.

Reply:
Fixed, adopted the same strategy as `ArithmeticOperation#swapLeftAndRight()`.